### PR TITLE
Add documentation to handle Java preview features for building natives

### PR DESCRIPTION
**Pull request by galderz**
Based on the feedback in [this Zulip stream](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/.E2.9C.94.20jdk17.20graalvm.20--enable-preview).

### CODE REVIEW

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -220,6 +220,15 @@ cmd /c 'call "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\
 In addition to the regular files, the build also produces `target/getting-started-1.0.0-SNAPSHOT-runner`.
 You can run it using: `./target/getting-started-1.0.0-SNAPSHOT-runner`.
 
+[[graal-package-preview]]
+[NOTE]
+.Java preview features
+====
+Java code might that relies on preview features requires special attention.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc:227]:
```suggestion
Java code that relies on preview features requires special attention.
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 2]
```diff
@@ -220,6 +220,15 @@ cmd /c 'call "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\
...
+To produce a native executable, this means that the `--enable-preview` flag needs to be passed to the underlying native image invocation.
+You can do so by prepending the flag with `-J` and passing it as additional native builder argument: `-Dquarkus.native.additional-build-args=-J--enable-preview`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc:229]:
```suggestion
You can do so by prepending the flag with `-J` and passing it as additional native build argument: `-Dquarkus.native.additional-build-args=-J--enable-preview`.
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 3]
```diff
@@ -319,6 +328,15 @@ Alternatively, you can run your tests with: `./mvnw verify -Pnative -Dquarkus.te
 However, don't forget that when the native executable is built the `prod` profile is enabled.
 So, the profile you enable this way must be compatible with the produced executable.
 
+[[graal-test-preview]]
+[NOTE]
+.Java preview features
+====
+Java code might that relies on preview features requires special attention.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc:335]:
```suggestion
Java code that relies on preview features requires special attention.
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 4]
```diff
@@ -319,6 +328,15 @@ Alternatively, you can run your tests with: `./mvnw verify -Pnative -Dquarkus.te
...
+To test a native executable, this means that the `--enable-preview` flag needs to be passed to surefire plugin.
+Adding `<argLine>--enable-preview</argLine>` to surefire plugin `configuration` section is one way to do so.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc:337]:
```suggestion
Adding `<argLine>--enable-preview</argLine>` to the Surefire plugin `configuration` section is one way to do so.
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 5]
```diff
@@ -319,6 +328,15 @@ Alternatively, you can run your tests with: `./mvnw verify -Pnative -Dquarkus.te
 However, don't forget that when the native executable is built the `prod` profile is enabled.
 So, the profile you enable this way must be compatible with the produced executable.
 
+[[graal-test-preview]]
+[NOTE]
+.Java preview features
+====
+Java code might that relies on preview features requires special attention.
+To test a native executable, this means that the `--enable-preview` flag needs to be passed to surefire plugin.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc:336]:
```suggestion
To test a native executable, this means that the `--enable-preview` flag needs to be passed to the Surefire plugin.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good, thanks. I added a few suggestions.

**Comment by galderz** [comment]
@gsmet Fixed comments

**Review by geoand — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**